### PR TITLE
Implement IResearchLinkCoordinator::stats()

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

Implement IResearchLinkCoordinator::stats()
Make client tests for it

- [ ] :hankey: Bugfix
- [x] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/IResearch/IResearchLinkCoordinator.cpp [hunk 1]
```diff
@@ -140,6 +140,44 @@ Result IResearchLinkCoordinator::init(velocypack::Slice definition) {
...
+    labels.push_back('=');
+    labels.push_back('"');
+    labels.append(value);
+    labels.push_back('"');
+  };
+  addLabel("db", getDbName());
+  addLabel("view", getViewId());
+  addLabel("collection", getCollectionName());
+  auto getValue = [&](const std::string& key) {
+    if (auto it = metrics.find({key, labels}); it != metrics.end()) {
```
Comment (gnusi) [inline arangod/IResearch/IResearchLinkCoordinator.cpp:168]: it seems we do a lot of allocations here, can we improve it anyhow?

Comment (gnusi) [inline arangod/IResearch/IResearchLinkCoordinator.cpp:168]: it seems we do a lot of allocations here, can we improve it anyhow?

File: arangod/IResearch/IResearchLink.cpp [hunk 2]
```diff
@@ -517,14 +517,10 @@ std::string const& IResearchLink::getShardName() const noexcept {
 }
 
 std::string IResearchLink::getCollectionName() const {
-  if (ServerState::instance()->isDBServer()) {
-    return _meta._collectionName;
-  }
   if (ServerState::instance()->isSingleServer()) {
```
Comment (gnusi) [inline arangod/IResearch/IResearchLink.cpp:520]: Let's leave assertion
```
TRI_ASSERT(!ServerState::instance()->isCoordinator());
```

Comment (gnusi) [inline arangod/IResearch/IResearchLink.cpp:520]: Let's leave assertion
```
TRI_ASSERT(!ServerState::instance()->isCoordinator());
```

Comment (MBkkt) [reply arangod/IResearch/IResearchLink.cpp:520]: Its not true

File: arangod/IResearch/IResearchLinkCoordinator.cpp (hunk 1 above)
Comment (MBkkt) [reply arangod/IResearch/IResearchLinkCoordinator.cpp:168]: I think I should use something like 
`absl::flat_hash_map<std::string_view, absl::flat_hash_map<std::string, Value>>`
Also I think we can make labels once in ctor?
In that case no allocation

File: arangod/IResearch/IResearchLink.cpp (hunk 2 above)
Comment (gnusi) [reply arangod/IResearch/IResearchLink.cpp:520]: Got it

File: tests/js/common/aql/aql-view-arangosearch-ddl-cluster.js [hunk 3]
```diff
@@ -26,8 +26,10 @@
 
 var jsunity = require("jsunity");
 var db = require("@arangodb").db;
+var isServer = require("@arangodb").isServer;
```
Comment (Dronplane) [inline tests/js/common/aql/aql-view-arangosearch-ddl-cluster.js:29]: NIT
```suggestion
const isServer = require("@arangodb").isServer;
```

**Review by Dronplane — APPROVED** [review]
LGTM

**Review by gnusi — APPROVED** [review]

**Comment by kapilpipaliya** [comment]
I can't compile.

**Comment by MBkkt** [comment]
@kapilpipaliya Jenkins builds successful :(

**Pull request merged**